### ISSUE TITLE
(Problem) Adding a real hardware interface to a host

### ISSUE THREAD

**Issue by hmd1st** [mininet/mininet#706]
### Expected/Desired Behavior:
Detach real hardware network interface from host in namespace
### Actual Behavior:
 real hardware network interface disappeared in ifconfig command
### Detailed Steps to Reproduce the Behavior:
Hello,
I have successfully added my hardware ethernet interface to a host in namespace=True and with this private directory: '/var/log'  
by this method:
_intf = Intf( intfName, node=h1) 
But after I closed mininet and cleaned the garbages by 'mn -c'. 
I realized that my ethernet interface has removed and it has disappeared in ifconfig and ip l commands but it still exists in lshw command. it seems the ethernet interface joins to host h1 namespace, and it cannot detach from h1 namespace by deleting that namespace.
I tried h1.delIntf(intfName) and _intf.delete() and h1.deletIntfs(). but none of them was not helpful.
Finally, when I restart my PC it solved. bud I am looking for a better solution. It is not wisely to restart PC every single time.
I am unsure if it is a bug in Intf or mn -c or not.

By the way, I appreciate any idea about that.

Best Regards,
Hamed

**Comment by lantz** [comment]
Can you include the complete code/test case?

Also, what is the interface name? By default, `mn -c` removes interfaces of the form `hN-ethY`, etc..

**Comment by hmd1st** [comment]
you can find a simple scenario in attachment:
[Uploading Test_addintf2host.py.zip…]()

Well, the interface name is not in the form of host interfaces like hN-ethY. this is simple ethX (for me eth0).
And finally, sorry for my poor english. I meant I stopped mininet running.
I myself add moveintf2rootns() method to node class for returning the interface to root ns by this command "ip link set intf.name netns 1" in that. but the main issue here is: if an error occurs after running this command "_intf = Intf( intfName, node=h1)" and before moveintf2rootns(), the interface remained on host ns and stay removed from root ns.
I am looking for a way to killing the related process which holds this interface even after killing mininet or something like that. Is it clear?
Thank you in advance

**Comment by ashtonwebster** [comment]
I know this is old now, but I am having the same problem.  Any luck resolving this?

**Comment by ashtonwebster** [comment]
For anyone else looking, the workaround recommended by @hmd1st works, but it is worth noting the `ip link set <hwintf> netns 1` command has to be executed by the node which was assigned the hwintf.  Thanks for the recommendation @hmd1st.  Still, would be nice to not have to use this workaround and just have mininet reset the hardware interfaces to the default namespace when done.

**Comment by lantz** [comment]
Can you confirm that the name of the interface doesn't have a name matching the canonical Mininet interface names (e.g. `h1-eth0` etc.?) Otherwise it will be probably deleted by `mn -c`.

**Comment by ashtonwebster** [comment]
Yes, the interface name is eth2 in my case, as observed by running "h3 ifconfig".  This is the same as it was before running mininet.

**Comment by hmd1st** [comment]
It seems the problem has been resolved in recent updates dear @ashtonwebster. 
Thank you @lants.
